### PR TITLE
plugin.youtube: bug fix for get video ID from currentVideoEndpoint

### PR DESCRIPTION
**Pull request by jitingcn**

### CODE REVIEW

**Comment by jitingcn** [comment]
It should work for all YT live stream video like: https://www.youtube.com/user/Wimbledon/live

File: src/streamlink/plugins/youtube.py [hunk 1]
```diff
@@ -97,7 +97,8 @@ def parse_fmt_list(formatsmap):
     }
 )
 
-_ytdata_re = re.compile(r'window\["ytInitialData"\]\s*=\s*({.*?});', re.DOTALL)
+# _ytdata_re = re.compile(r'window\["ytInitialData"\]\s*=\s*({.*?});', re.DOTALL)
```
[the remark this answers is not in the document]
Comment (jitingcn) [reply src/streamlink/plugins/youtube.py]: Sorry, I forgot delete that.

**Comment by ErAzOr2k** [comment]
doesn't work for youtube.com/rocketbeanstv/live

**Comment by jitingcn** [comment]
> doesn't work for youtube.com/rocketbeanstv/live

That maybe another issue, seem can't get correctly hls url in best quality when 1080p and 1080p60 exist at the same time.

Try to download best quality (1080p60) with format code 299
```
$ streamlink -o test.ts https://www.youtube.com/rocketbeanstv/live best -l debug
[cli][debug] OS:         Linux-5.1.15-arch1-1-ARCH-x86_64-with-arch
[cli][debug] Python:     3.7.3
[cli][debug] Streamlink: 1.1.1+39.g04f0b0d
[cli][debug] Requests(2.22.0), Socks(1.6.7), Websocket(0.56.0)
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/rocketbeanstv/live
[plugin.youtube][debug] Video ID from currentVideoEndpoint
[plugin.youtube][debug] Using video ID: E-MsayZ_Epw
[plugin.youtube][debug] get_video_info - 1: Found data
[plugin.youtube][debug] MuxedStream: v 137 a 140 = 1080p
[plugin.youtube][debug] MuxedStream: v 299 a 140 = 1080p60
[utils.l10n][debug] Language code: zh_CN
[cli][info] Available streams: audio_mp4, 144p (worst), 240p, 360p, 480p, 720p, 1080p, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (muxed-stream)
[stream.ffmpegmux][debug] Opening http substream
[stream.ffmpegmux][debug] Opening http substream
[stream.mp4mux-ffmpeg][debug] ffmpeg command: ffmpeg -nostats -y -i /tmp/ffmpeg-74555-150 -i /tmp/ffmpeg-74555-765 -c:v copy -c:a copy -map 0 -map 1 -f matroska pipe:1
[stream.ffmpegmux][debug] Starting copy to pipe: /tmp/ffmpeg-74555-150
[stream.ffmpegmux][debug] Starting copy to pipe: /tmp/ffmpeg-74555-765
[cli][debug] Pre-buffering 8192 bytes
[stream.ffmpegmux][debug] Pipe copy complete: /tmp/ffmpeg-74555-765
[cli][debug] Checking file output
[cli][debug] Writing stream to output
[stream.ffmpegmux][debug] Pipe copy complete: /tmp/ffmpeg-74555-150

[stream.ffmpegmux][debug] Closing ffmpeg thread
[stream.ffmpegmux][debug] Closed all the substreams
[cli][info] Stream ended
[cli][info] Closing currently open stream...
[stream.ffmpegmux][debug] Closing ffmpeg thread
[stream.ffmpegmux][debug] Closed all the substreams
[stream.ffmpegmux][debug] Closing ffmpeg thread
[stream.ffmpegmux][debug] Closed all the substreams
```
It will try to get the http substream and exit directly, only one frame/segment of data can be obtained.

If you choose 1080p quality, it can download as normal.
```
$ streamlink -o test.ts https://www.youtube.com/rocketbeanstv/live 1080p -l debug
[cli][debug] OS:         Linux-5.1.15-arch1-1-ARCH-x86_64-with-arch
[cli][debug] Python:     3.7.3
[cli][debug] Streamlink: 1.1.1+39.g04f0b0d
[cli][debug] Requests(2.22.0), Socks(1.6.7), Websocket(0.56.0)
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/rocketbeanstv/live
[plugin.youtube][debug] Video ID from currentVideoEndpoint
[plugin.youtube][debug] Using video ID: E-MsayZ_Epw
[plugin.youtube][debug] get_video_info - 1: Found data
[plugin.youtube][debug] MuxedStream: v 137 a 140 = 1080p
[plugin.youtube][debug] MuxedStream: v 299 a 140 = 1080p60
[utils.l10n][debug] Language code: zh_CN
[cli][info] Available streams: audio_mp4, 144p (worst), 240p, 360p, 480p, 720p, 1080p, 1080p60 (best)
[cli][info] Opening stream: 1080p (hls)
[stream.hls][debug] Reloading playlist
[stream.hls][debug] First Sequence: 72157; Last Sequence: 75755
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 75753; End Sequence: None
[stream.hls][debug] Adding segment 75753 to queue
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Adding segment 75754 to queue
[stream.hls][debug] Adding segment 75755 to queue

```

File: src/streamlink/plugins/youtube.py [hunk 2]
```diff
@@ -272,7 +273,7 @@ def _find_video_id(self, url):
         res = self.session.http.get(url)
         datam = _ytdata_re.search(res.text)
         if datam:
-            data = parse_json(datam.group(1))
+            data = parse_json(json.loads(datam.group(1)))
```
Comment (back-to) [inline src/streamlink/plugins/youtube.py]:
```suggestion
            data = parse_json(parse_json(datam.group(1)))
```

would also work

**Comment by jitingcn** [comment]
Now YouTube change back the pattern of content 'window["ytInitialData"]', I will close this pr.

**Pull request closed without merging**